### PR TITLE
PVP-186

### PR DESCRIPTION
**Pull request by KhajiitLikeToSneak**
Improved filtering design (editing and surveys)

### CODE REVIEW

File: app/src/main/java/com/pvp/app/ui/screen/filters/FiltersComponents.kt [hunk 1]
```diff
@@ -114,51 +130,84 @@ fun FiltersBox(
...
+                    text = boxTitle,
                     style = TextStyle(fontSize = 18.sp),
                     modifier = Modifier.padding(8.dp)
                 )
             }
 
+            if (filters.isEmpty()) {
+                Text(
+                    text = if (isSelected == false) "No other $title" else "No $title selected",
+                    style = TextStyle(fontSize = 15.sp, fontStyle = FontStyle.Italic),
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/filters/FiltersComponents.kt:144]: Split params.

File: app/src/main/java/com/pvp/app/ui/screen/filters/FiltersComponents.kt [hunk 2]
```diff
@@ -114,51 +130,84 @@ fun FiltersBox(
...
+                                    text = filter,
+                                    style = TextStyle(fontSize = 15.sp),
+                                )
+                            }
+
+                            if (isSelected != null) {
+                                Box(
+                                    modifier = Modifier
+                                        .align(Alignment.TopEnd)
+                                        .padding(end = 0.dp, top = 0.dp)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/filters/FiltersComponents.kt:194]: Params.

File: app/src/main/java/com/pvp/app/ui/screen/survey/Surveys.kt [hunk 3]
```diff
@@ -211,11 +245,19 @@ fun FiltersBox(
...
-                style = TextStyle(fontSize = 18.sp),
                 modifier = Modifier.padding(8.dp),
-                text = title
+                style = TextStyle(fontSize = 18.sp),
+                text = boxTitle
             )
 
+            if (filters.isEmpty()) {
+                Text(
+                    text = if (isSelected == false) "No other $title" else "No $title selected",
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/survey/Surveys.kt:255]: `isSelected` is a non-nullable boolean here.

```kotlin
if (!isSelected) ...
```
I believe there was another component like this one, where boolean was nullable. If you can evade nullability, then evade it. But if nullable was there to indicate some sort of state in which different components are shown, let it be.

File: app/src/main/java/com/pvp/app/ui/screen/survey/Surveys.kt [hunk 4]
```diff
@@ -211,11 +245,19 @@ fun FiltersBox(
...
+                    style = TextStyle(fontSize = 15.sp, fontStyle = FontStyle.Italic),
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/survey/Surveys.kt:256]: Params.

File: app/src/main/java/com/pvp/app/ui/screen/survey/Surveys.kt [hunk 5]
```diff
@@ -224,20 +266,58 @@ fun FiltersBox(
...
+                                        ),
+                                    style = TextStyle(fontSize = 15.sp),
+                                    text = it
+                                )
+                            }
+
+                            Box(
+                                modifier = Modifier
+                                    .align(Alignment.TopEnd)
+                                    .padding(end = 0.dp, top = 0.dp)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/survey/Surveys.kt:305]: Params.

File: app/src/main/java/com/pvp/app/ui/screen/survey/Surveys.kt [hunk 6]
```diff
@@ -224,20 +266,58 @@ fun FiltersBox(
...
+                                    .background(Color.White)
+                                    .border(
+                                        width = 1.dp,
+                                        color = MaterialTheme.colorScheme.secondary,
+                                        shape = CircleShape
+                                    )
+                                    .size(16.dp)
+                            ) {
+                                Icon(
+                                    imageVector = if (isSelected == true) Icons.Outlined.Remove else Icons.Outlined.Add,
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/survey/Surveys.kt:316]: Once again, boolean here is non-nullable. `if (isSelected)` is enough.

**Review by klaidoshka — APPROVED** [review]

**Review by vytjar — APPROVED** [review]

**Pull request merged**